src/site/content/en/blog/smoothness/index.md

```diff
@@ -0,0 +1,563 @@
+efficiently render and present visual updates!
+
+Some examples:
+
+- Using content that is too large or resource intensive to decode quickly on the
+  target device.
+- [Using too many
+  layers](https://developers.google.com/web/fundamentals/performance/rendering/stick-to-compositor-only-properties-and-manage-layer-count)
+  requiring too much GPU memory.
+- Defining overly-complex CSS styles or web animations.
+- Using design anti-patterns that disable fast rendering optimizations.
+- Too much JS work on the main thread, leading to long tasks that block certain
+  visual updates.
+
+But how can you know when an animation frame has slipped its deadline and caused
+a dropped frame?
+
+You may be aware of one method:
+[`requestAnimationFrame()`](https://developer.mozilla.org/docs/Web/API/window/requestAnimationFrame)
+polling. `requestAnimationFrame()`, or "rAF", tells the browser that you wish to
+perform an animation and asks for an opportunity to do so before the next paint
+stage of the rendering pipeline. If your callback function isn't called at the
+time you expect it, that means a paint wasn't executed, and one or more frames
+were skipped. By polling and counting how often rAF is called, you can compute a
+sort of "frames per second" (fps) metric.
+
+```js
+let frameTimes = [];
+function pollFramesPerSecond(now) {
+  frameTimes = [...frameTimes.filter(t => t > now - 1000), now];
+  requestAnimationFrame(pollFramesPerSecond);
+  console.log('Frames per second:', frameTimes.length);
+}
+requestAnimationFrame(pollFramesPerSecond);
+```
+
+{% Aside 'warning' %}
+  The code above is an anti pattern and is strongly discouraged!
+{% endAside %}
+
+Using `requestAnimationFrame()` polling is not a good idea for several reasons:
+
+- Every interested party/script has to set up its own polling loop.
+- It can block the critical path.
+- Even if the rAF polling function is "fast", when used continuously it can
+  prevent `requestIdleCallback()` from being able to schedule long idle blocks
+  (i.e. blocks that exceed a single frame).
+- Similarly, lack of long idle-blocks prevents the browser from scheduling other
+  long running tasks (e.g. longer GC's and other background or speculative
+  work).
+- If polling is toggled on/off, then you'll miss cases where frame budget has
+  been exceeded.
+- Polling will report false-positives in cases where the browser is using
+  variable update frequency (e.g. due to power, visibility status, etc.).
+- And most importantly, it doesn't actually capture all types of animation
+  updates!
+
+You can see how too much work on the main thread can impact the ability to see
+animation frames. You can play with the [Jank
+Sample](https://googlechrome.github.io/devtools-samples/jank/) to see how a rAF
+driven animation, once there is too much work on the main thread work (i.e.
+layout) will lead to dropped frames and fewer rAF callbacks, and lower fps.
```

Comment:
```suggestion
Too much work on the main thread can impact the ability to see
animation frames. Check out the [Jank
Sample](https://googlechrome.github.io/devtools-samples/jank/) to see how a rAF
driven animation, once there is too much work on the main thread (such as
layout), will lead to dropped frames and fewer rAF callbacks, and lower FPS.
```

Reply:
I'd add instructions here to open the Rendering performance monitor in DevTools. Or a link to instructions.